### DISCUSSION TITLE
What is the purpose of registerObservers?

### DISCUSSION THREAD

**Discussion by francoism90** [Safemood/laravel-workflow#5] [General]
### What happened?

You don't have discussions enabled (yet), sorry for this being reported as bug. :)

Your package looks very interesting, but I do have a few questions:
- What's the purpose of registerObservers? Are they used for temporary tracking? I would expect them to be globally defined instead. Do you have a good example?
- Giving the example `$paymentWorkflow = (new PaymentWorkflow)->run($context);`, how can you access `$context`? Or is this being injected into every action call?
- `class PaymentWorkflow extends Workflow`, I don't see the `Workflow` class?

Thanks!

### How to reproduce the bug

N/A

### Package Version

1.0.0

### PHP Version

8.3.10

### Laravel Version

11.0

### Which operating systems does with happen with?

Linux

**Comment by Safemood** [comment]
Hi @francoism90! Thanks for checking out the package and sharing your questions!

Thanks for noticing, we now have Discussions up and running! 😊🎉

**registerObservers**: You can define them manually within the workflow with this method to keep everything in one place, or use the CLI to auto-register them. Then, track observer events using methods like trackAllEvents.

Accessing $context: The $context is automatically injected into each action, so you can access it seamlessly in other parts of the workflow.

Workflow class: You’re absolutely right there’s a typo in the docs. It should be WorkflowManager. Thanks for pointing that out!

Let me know if you have any more questions or need further clarification!
